provisioning/device/test/_symmetric_key_registration_test.js

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var sinon = require('sinon');
+var assert = require('chai').assert;
+
+var SymmetricKeyRegistration = require('../lib/symmetric_registration').SymmetricKeyRegistration;
+
+var fakeProvisioningHost = 'fakeHost';
+var fakeIdScope = 'fakeIdScope';
+var fakeRegistrationId = 'fakeRegistrationId';
+var fakeSasToken = 'a fake token';
+
+var fakeResponse = {
+  assignedHub: 'fakeHub',
+  deviceId: 'fakeDeviceId'
+};
+
+describe('Symmetric Key Registration', function () {
+  describe('#register', function () {
+
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_006: [ `register` shall call the `getRegistrationId` method on the security object to acquire the registration id. ] */
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_008: [ `register` shall invoke `createAuthenticationToken` method on the security object to acquire a sas token object. ] */
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_004: [ `register` shall pass the SAS token into the `setSasToken` method on the transport. ] */
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_005: [ `register` shall call `register` on the polling state machine object. ] */
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_011: [ Otherwise `register` shall invoke the `_callback` with the resultant `registrationState` as the second argument. ] */
+    it('gets the registration id and consequent sas token and registers', function(callback) {
```

Comment:
>gets the registration id and consequent sas token and registers [](start = 8, length = 63)

nit: sounds like 3 different unit tests, if not 4.

Reply:
Happy path test.

Other tests that follow get the other paths.

---
In reply to: [231361906](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231361906) [](ancestors = 231361906)

Reply:
which is different from unit tests. now if I break something and that test fails, I don't know what i broke. For happy paths, we have e2e tests.

---
In reply to: [231620930](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231620930) [](ancestors = 231620930,231361906)

Reply:
And the same as other security client tests.   Err, well if this test fails you have something to work with.  Are you actually suggesting I simply take this out?  I disagree.

---
In reply to: [231751816](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231751816) [](ancestors = 231751816,231620930,231361906)

Reply:
Done.

---
In reply to: [232118379](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r232118379) [](ancestors = 232118379,232040961,231751816,231620930,231361906)